package.json

```diff
@@ -30,7 +30,8 @@
     "build-emscripten": "yarn build-update-h3 && yarn docker-emscripten",
     "build-legacy": "node scripts/build-legacy-api.js",
     "build-docs": "jsdoc2md --no-cache --global-index-format grouped --partial doc-files/scope.hbs --helper ./doc-files/insert-version.js --separators --template doc-files/README.tmpl.md lib/h3core.js lib/errors.js > README.md",
-    "build-tsd-core": "jsdoc -t node_modules/tsd-jsdoc/dist -d console lib/h3core.js | sed 's/\"h3\"/\"h3-js\"/g' > dist/types.d.ts",
+    "build-tsd-core": "tsc lib/h3core.js --noResolve --skipLibCheck --allowJs --declaration --emitDeclarationOnly --outFile dist/types.d.ts && yarn build-tsd-core-postprocess",
+    "build-tsd-core-postprocess": "sed 's/module \"h3core\"/module \"h3-js\"/g' dist/types.d.ts > types.d.ts.tmp && sed -E -e 's/(CoordPair|SplitLong) = number\\[\\]/\\1 = [number,number]/g' types.d.ts.tmp > dist/types.d.ts && rm types.d.ts.tmp",
```

Comment:
Maybe this logic should be in a script file so that comments can be added to it? This is a little 'magical'

Reply:
Agree, I can take a look at moving this into a script